releasenotes/notes/rust-two-qubit-weyl-ec551f3f9c812124.yaml

```diff
@@ -0,0 +1,14 @@
+---
+features_synthesis:
+  - |
+    The :class:`.TwoQubitWeylDecomposition` synthesis class has been rewritten
+    in Rust for better performance.
+upgrade_synthesis:
+  - |
+    The :class:`.TwoQubitWeylDecomposition` no longer will self specialize into
+    a subclass on creation. This was an internal detail of the :class:`.TwoQubitWeylDecomposition`
+    previously, and was not a documented public behavior as all the subclasses behaved
+    the same and were only used for internal dispatch. However, as it was discoverable
+    behavior this release note is to document that this will no longer occur and all
+    instances of :class:`.TwoQubitWeylDecomposition` will be of the same type. There is no
+    change in behavior for public methods of the class.
```

Comment:
Technically, `TwoQubitWeylDecomposition.specialize` was public before and now doesn't exist.  I mean, it should never have been called by a user, but it technically was public.

(Also, "no longer will self specialize" is a bit of an awkward word order - "will not longer self-specialize" might be a bit easier to parse.)

Reply:
Oh, that's a good call. We should add it back with the default `NotImplementedError` raise because it no longer does anything. This isn't a change in behavior because we're not subclassing anymore and the default for `TwoQubitWeylDecomposition` was already to raise.

Reply:
Added the public method back and deprecated it in: https://github.com/Qiskit/qiskit/pull/11946/commits/1a8be8549588fdbeab259e6db5070e016439e086